pkg/controller/versions/strategy_v2_1.go

```diff
@@ -266,6 +271,46 @@ func (v *versionStrategyV2_1) Render(ctx context.Context, cr *common.ControllerR
 		return nil, fmt.Errorf("Could not set field status.lastAppliedConfiguration.istio.istio_cni.istio_cni_network: %v", err)
 	}
 
+	if smcp.Spec.ControlPlaneMode != nil {
+		if smcp.Spec.ControlPlaneMode.Enabled != nil {
+			err = spec.Istio.SetField("istioDiscovery.enabled", *smcp.Spec.ControlPlaneMode.Enabled)
+		} else {
+			err = spec.Istio.SetField("istioDiscovery.enabled", true)
+		}
+	}
+
+	if smcp.Spec.Telemetry != nil {
+		if smcp.Spec.Telemetry.Enabled != nil {
+			err = spec.Istio.SetField("telemetry.common.enabled", *smcp.Spec.Telemetry.Enabled)
+		} else {
+			err = spec.Istio.SetField("telemetry.common.enabled", true)
+		}
+	}
+
+	if smcp.Spec.RemoteMode != nil {
```

Comment:
won't you need to set `remotePilotAddress` in this case?

Reply:
We didn't need that for our setup but from the chart install enablement perspective it is not relevant